### DISCUSSION TITLE
Deploiement SORMAS en Production

### DISCUSSION THREAD

**Discussion by arigui** [SORMAS-Foundation/SORMAS-Project#12985] [Technical Questions]
Bonjour,
dans le cadre de découvrir la solution SORMAS, je décide de l'installer en prod sur un serveur Ubuntu 22.04 LTS en suivant les [instructions données](https://github.com/SORMAS-Foundation/SORMAS-Project/blob/development/docs/SERVER_SETUP.md).
Après l'exécution de ./server-setup.sh, j'ai accès à la console d'administration payara mais je n'arrive pas à déployer sormas-ear, sormas-rest, sormas-ui.
Pouvez vous m'aider à ressourdre le problème?

ci dessous une vue de server.log

[2024-02-19T11:16:03.103+0000] [Payara 5.2022.5] [INFOS] [] [org.glassfish.admingui] [tid: _ThreadID=155 _ThreadName=admin-thread-pool::admin-listener(9)] [timeMillis: 17    08341363103] [levelValue: 800] [[
270   GUI deployment: uploadToTempfile]]
271
272 [2024-02-19T11:16:03.129+0000] [Payara 5.2022.5] [INFOS] [] [org.glassfish.admingui] [tid: _ThreadID=155 _ThreadName=admin-thread-pool::admin-listener(9)] [timeMillis: 17    08341363129] [levelValue: 800] [[
273   uploadFileName=sormas-ear.ear]]
274
275 [2024-02-19T11:16:03.164+0000] [Payara 5.2022.5] [INFOS] [] [] [tid: _ThreadID=155 _ThreadName=admin-thread-pool::admin-listener(9)] [timeMillis: 1708341363164] [levelVal    ue: 800] [[
276   ================== availabilityEnabled  skipped ]]
277
278 [2024-02-19T11:16:08.716+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368716] [levelValue: 1000] [[
279   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.campaign.form.CampaignFormMetaService ]]]
280
281 [2024-02-19T11:16:08.744+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368744] [levelValue: 1000] [[
282   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.event.EventService ]]]
283
284 [2024-02-19T11:16:08.875+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368875] [levelValue: 1000] [[
285   Class [ org/apache/commons/lang3/NotImplementedException ] not found. Error while loading [ class de.symeda.sormas.backend.environment.EnvironmentFacadeEjb$EnvironmentF    acadeEjbLocal ]]]
286
287 [2024-02-19T11:16:08.882+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368882] [levelValue: 1000] [[
288   Class [ org/hibernate/transform/ResultTransformer ] not found. Error while loading [ class de.symeda.sormas.backend.immunization.DirectoryImmunizationService ]]]
289
290 [2024-02-19T11:16:08.896+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368896] [levelValue: 1000] [[
291   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.sormastosormas.entities.contact.SormasToSormasContactFacadeEjb$SormasToSorm    asContactFacadeEjbLocal ]]]
292
293 [2024-02-19T11:16:08.901+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368901] [levelValue: 1000] [[
294   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.environment.EnvironmentService ]]]
295
296 [2024-02-19T11:16:08.919+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368919] [levelValue: 1000] [[
297   Class [ org/apache/commons/lang3/NotImplementedException ] not found. Error while loading [ class de.symeda.sormas.backend.travelentry.TravelEntryFacadeEjb$TravelEntryF    acadeEjbLocal ]]]
298
[2024-02-19T11:16:08.922+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368922] [levelValue: 1000] [[
300   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.deletionconfiguration.DeletionConfigurationService ]]]
301
302 [2024-02-19T11:16:08.936+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368936] [levelValue: 1000] [[
303   Class [ org/apache/commons/lang3/NotImplementedException ] not found. Error while loading [ class de.symeda.sormas.backend.campaign.CampaignFacadeEjb$CampaignFacadeEjbL    ocal ]]]
304
305 [2024-02-19T11:16:08.982+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368982] [levelValue: 1000] [[
306   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.environment.EnvironmentImportFacadeEjb$EnvironmentImportFacadeEjbLocal ]]]
307
308 [2024-02-19T11:16:08.991+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368991] [levelValue: 1000] [[
309   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.event.EventGroupFacadeEjb$EventGroupFacadeEjbLocal ]]]
310
311 [2024-02-19T11:16:08.992+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368992] [levelValue: 1000] [[
312   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.environment.environmentsample.EnvironmentSampleService ]]]
313
314 [2024-02-19T11:16:08.999+0000] [Payara 5.2022.5] [GRAVE] [] [Annotation-Framework] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMillis: 1708    341368999] [levelValue: 1000] [[
315   Class [ Lorg/slf4j/Logger; ] not found. Error while loading [ class de.symeda.sormas.backend.travelentry.travelentryimport.TravelEntryImportFacadeEjb$TravelEntryImportF    acadeEjbLocal ]]]
316
317 [2024-02-19T11:16:09.040+0000] [Payara 5.2022.5] [GRAVE] [] [javax.enterprise.system.tools.deployment.dol] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listen    er(7)] [timeMillis: 1708341369040] [levelValue: 1000] [[
318   org/apache/commons/lang3/NotImplementedException]]
319
320 [2024-02-19T11:16:09.044+0000] [Payara 5.2022.5] [GRAVE] [] [javax.enterprise.system.core] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listener(7)] [timeMill    is: 1708341369044] [levelValue: 1000] [[
321   Exception while deploying the app [sormas-ear]]]
[2024-02-19T11:16:09.046+0000] [Payara 5.2022.5] [GRAVE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=149 _ThreadName=admin-thread-pool::admin-listene    r(7)] [timeMillis: 1708341369046] [levelValue: 1000] [[
324   Exception during lifecycle processing
325 java.lang.IllegalStateException: org/apache/commons/lang3/NotImplementedException. Related annotation information: annotation [@javax.annotation.security.PermitAll()] on     annotated element [public de.symeda.sormas.api.sormastosormas.share.outgoing.SormasToSormasShareInfoDto de.symeda.sormas.backend.sormastosormas.share.outgoing.SormasToSor    masShareInfoFacadeEjb.toDto(de.symeda.sormas.backend.sormastosormas.share.outgoing.SormasToSormasShareInfo)] of type [METHOD]
326         at com.sun.enterprise.deployment.archivist.Archivist.readAnnotations(Archivist.java:542)
327         at com.sun.enterprise.deployment.archivist.Archivist.readAnnotations(Archivist.java:459)
328         at com.sun.enterprise.deployment.archivist.Archivist.readRestDeploymentDescriptors(Archivist.java:429)
329         at com.sun.enterprise.deployment.archivist.Archivist.readDeploymentDescriptors(Archivist.java:395)
330         at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:271)
331         at com.sun.enterprise.deployment.archivist.Archivist.open(Archivist.java:280)
332         at com.sun.enterprise.deployment.archivist.ApplicationArchivist.readModulesDescriptors(ApplicationArchivist.java:629)
333         at com.sun.enterprise.deployment.archivist.ApplicationArchivist.openWith(ApplicationArchivist.java:249)
334         at com.sun.enterprise.deployment.archivist.ApplicationFactory.openWith(ApplicationFactory.java:232)
335         at org.glassfish.javaee.core.deployment.DolProvider.processDOL(DolProvider.java:211)
336         at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:246)
337         at org.glassfish.javaee.core.deployment.DolProvider.load(DolProvider.java:97)
338         at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployer(ApplicationLifecycle.java:1101)
339         at com.sun.enterprise.v3.server.ApplicationLifecycle.loadDeployers(ApplicationLifecycle.java:1069)
340         at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:884)
341         at com.sun.enterprise.v3.server.ApplicationLifecycle.prepare(ApplicationLifecycle.java:452)
342         at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:593)
343         at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:556)
344         at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:552)
345         at java.base/java.security.AccessController.doPrivileged(Native Method)
346         at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
347         at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:551)
348         at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:582)
349         at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:574)
350         at java.base/java.security.AccessController.doPrivileged(Native Method)
351         at java.base/javax.security.auth.Subject.doAs(Subject.java:361)
352         at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:573)
353         at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1497)
354         at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1200(CommandRunnerImpl.java:120)
355         at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1869)
356         at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1755)
357         at org.glassfish.admin.rest.utils.ResourceUtil.runCommand(ResourceUtil.java:272)
358         at org.glassfish.admin.rest.utils.ResourceUtil.runCommand(ResourceUtil.java:240)
359         at org.glassfish.admin.rest.utils.ResourceUtil.runCommand(ResourceUtil.java:294)
360         at org.glassfish.admin.rest.resources.TemplateListOfResource.createResource(TemplateListOfResource.java:136)
361         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
362         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
363         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                                                                                                                                                          
lterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
534         at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
535         at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
536         at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
537         at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
538         at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
539         at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
540         at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
541         at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
542         at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
543         at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
544         at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
545         at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
546         at java.base/java.lang.Thread.run(Thread.java:829)
547 Caused by: java.lang.NoClassDefFoundError: org/apache/commons/lang3/NotImplementedException
548         at java.base/java.lang.Class.forName0(Native Method)
549         at java.base/java.lang.Class.forName(Class.java:398)
550         at com.sun.enterprise.deployment.util.TypeUtil.getMethod(TypeUtil.java:397)
551         at com.sun.enterprise.deployment.MethodDescriptor.getMethod(MethodDescriptor.java:291)
552         at com.sun.enterprise.deployment.annotation.handlers.AbstractAuthAnnotationHandler.processAnnotation(AbstractAuthAnnotationHandler.java:115)
553         at com.sun.enterprise.deployment.annotation.handlers.AbstractCommonAttributeHandler.processAnnotation(AbstractCommonAttributeHandler.java:153)
554         at com.sun.enterprise.deployment.annotation.factory.SJSASFactory$LazyAnnotationHandler.processAnnotation(SJSASFactory.java:170)
555         at org.glassfish.apf.impl.AnnotationProcessorImpl.process(AnnotationProcessorImpl.java:375)
556         ... 97 more
557 Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.NotImplementedException
558         at org.glassfish.internal.api.DelegatingClassLoader.findClass(DelegatingClassLoader.java:193)
559         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:594)
560         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:527)
561         at com.sun.enterprise.loader.CurrentBeforeParentClassLoader.loadClass(CurrentBeforeParentClassLoader.java:98)
562         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:527)
563         ... 105 more
564 ]]
565
566 [2024-02-19T11:16:09.156+0000] [Payara 5.2022.5] [INFOS] [] [org.glassfish.admingui] [tid: _ThreadID=155 _ThreadName=admin-thread-pool::admin-listener(9)] [timeMillis: 17    08341369156] [levelValue: 800] [[
567   Exception Occurred :Error occurred during deployment: Exception while deploying the app [sormas-ear] : org/apache/commons/lang3/NotImplementedException. Related annotat    ion information: annotation [@javax.annotation.security.PermitAll()] on annotated element [public de.symeda.sormas.api.sormastosormas.share.outgoing.SormasToSormasShareIn    foDto de.symeda.sormas.backend.sormastosormas.share.outgoing.SormasToSormasShareInfoFacadeEjb.toDto(de.symeda.sormas.backend.sormastosormas.share.outgoing.SormasToSormasS    hareInfo)] of type [METHOD]. Please see server.log for more details. ]]

**Comment by edlixstudios** [comment]
Bonjour @arigui  j'ai le même problème. As-tu déjà trouvé une solution ?

**Reply by arigui** [reply]
Bonjour @edlixstudios, pas encore.

**Comment by Candice-Louw** [comment]
Bonjour @arigui , @edlixstudios ,
Pourriez-vous s'il vous plaît poster cette question sur la page de la Communauté SORMAS ici : https://community.sormas.org/

Cela permettra à un plus grand nombre d'utilisateurs actifs de voir et de répondre à votre question afin que nous puissions la résoudre. Il existe un canal dédié aux questions en français, que vous pouvez utiliser si vous voulez : https://community.sormas.org/c/sormas-francais

Cordialement!
